test/e2e/create_enable_disable_delete.go

```diff
@@ -12,68 +12,186 @@ import (
+			verifier.verifyThatLogsAreNotSentToEchoServer(ctx, "test-program", "3", "this should not get sent to echo server")
+
+			By("Disable the shoot-rsyslog-relp extension")
+			ctx, cancel = context.WithTimeout(parentCtx, 15*time.Minute)
+			DeferCleanup(cancel)
+			Expect(f.UpdateShoot(ctx, f.Shoot, func(shoot *gardencorev1beta1.Shoot) error {
+				for i, extension := range shoot.Spec.Extensions {
+					if extension.Type == "shoot-rsyslog-relp" {
+						shoot.Spec.Extensions = append(shoot.Spec.Extensions[:i], shoot.Spec.Extensions[i+1:]...)
+					}
 				}
+				return nil
+			})).To(Succeed())
+
+			By("Verify that shoot-rsyslog-relp extension is disabled")
+			ctx, cancel = context.WithTimeout(parentCtx, 5*time.Minute)
+			DeferCleanup(cancel)
+			verifier.verifyThatLogsAreNotSentToEchoServer(ctx, "test-program", "1", "this should not get sent to echo server")
+
+			By("Delete Shoot")
+			ctx, cancel = context.WithTimeout(parentCtx, 15*time.Minute)
+			DeferCleanup(cancel)
+			Expect(f.DeleteShootAndWaitForDeletion(ctx, f.Shoot)).To(Succeed())
+
+			By("Delete additional garden cluster resources")
+			for _, obj := range gardenClusterResources {
+				Expect(f.GardenClient.Client().Delete(ctx, obj)).To(Succeed())
+				f.Logger.Info("Deleted resource", "resource", client.ObjectKeyFromObject(obj))
 			}
-			return nil
-		})).To(Succeed())
+		})
+	}
 
-		By("Verify that shoot-rsyslog-relp extension is disabled")
-		ctx, cancel = context.WithTimeout(parentCtx, 5*time.Minute)
-		defer cancel()
-		verifier.verifyThatLogsAreNotSentToEchoServer(ctx, "test-program", "1", "this should not get sent to echo server")
+	Context("shoot-rsyslog-relp extension with tls disabled", Label("tls-disabled"), func() {
+		test(e2e.DefaultShoot("e2e-rslog-relp"), func(shoot *gardencorev1beta1.Shoot) error {
+			shoot.Spec.Extensions = append(shoot.Spec.Extensions, shootRsyslogRelpExtension())
+			return nil
+		})
+	})
 
-		By("Delete Shoot")
-		ctx, cancel = context.WithTimeout(parentCtx, 15*time.Minute)
-		defer cancel()
-		Expect(f.DeleteShootAndWaitForDeletion(ctx, f.Shoot)).To(Succeed())
+	Context("shoot-rsyslog-relp extension with tls enabled", Label("tls-enabled"), func() {
+		test(
+			e2e.DefaultShoot("e2e-rslog-tls"),
+			func(shoot *gardencorev1beta1.Shoot) error {
+				shoot.Spec.Extensions = append(shoot.Spec.Extensions, shootRsyslogRelpExtension(withPort(443), withTLSWithSecretRefName("rsyslog-relp-tls")))
+				shoot.Spec.Resources = append(shoot.Spec.Resources, gardencorev1beta1.NamedResourceReference{
+					Name: "rsyslog-relp-tls",
+					ResourceRef: autoscalingv1.CrossVersionObjectReference{
+						Kind:       "Secret",
+						APIVersion: "v1",
+						Name:       "rsyslog-relp-tls",
+					},
+				})
+				return nil
+			},
+			&corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "rsyslog-relp-tls",
+				},
+				Data: map[string][]byte{
+					"ca": []byte(`-----BEGIN CERTIFICATE-----
```

Comment:
Idea: You could move it to files under `$PWD/testdata` and load them with `go:embed` for the test? WDYT?

Reply:
Thanks for the suggestion, I found a function in `"github.com/gardener/gardener/pkg/utils/test.EnsureTestResources` so I can add the whole resource in a `.yaml` file and use that function to deploy it